### ISSUE TITLE
no-restricted-modules doesn't check es6 modules

### ISSUE THREAD

**Issue by zaggino** [eslint/eslint#3196]
setting
```no-restricted-modules: [2, "cluster"]```
doesn't throw error on
```import cluster from "cluster";```
probably due to:
https://github.com/eslint/eslint/blob/2f36f92e1c8d7000252f905c0a90764ff06e7249/lib/rules/no-restricted-modules.js#L61

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by xjamundx** [comment]
Yeah that sounds like an oversight to me.

**Comment by nzakas** [comment]
Not an oversight - this was created for `require` syntax. We generally do not treat `require` the sand as ES6 modules because they behave differently.

**Comment by xjamundx** [comment]
@nzakas because it's called `no-restricted-modules`, would it be ok to add import declarations to the list of places that could trigger the warning. I don't think it makes sense for people to make that list twice.

**Comment by nzakas** [comment]
Wouldn't people be using one or the other? Not both?

**Comment by xjamundx** [comment]
Not in our case where we have a transitioning codebase, but I know you've
been consistent about this elsewhere so I'm fine with a new rule...

**Comment by michaelficarra** [comment]
I'd be really disappointed if this rule couldn't handle both styles.

**Comment by nzakas** [comment]
I'm just really worried about bundling these together when we don't know what format module strings will ultimately take. People "using" ES6 modules today are doing so built on the back of `require`, but there's no guarantee that module strings will be in the same format when native implementations arrive.

**Comment by xjamundx** [comment]
If *we* do make a new rule, I guess we'd have to create a new rule and then deprecated this one?

- `no-require-modules`
- `no-import-modules`

One argument in favor of making them separate: AMD modules. If we wanted to add a `no-define-modules` really it would be really easy to add that without having a fight as to if it's worth overloading the `no-restricted-modules` rule.

**Comment by nzakas** [comment]
No, I don't see any reason to deprecate the existing one, I'd just add `no-restricted-imports`

**Comment by gyandeeps** [comment]
I think the decision has been made and we need label updates?
CC @nzakas

**Comment by guyellis** [comment]
I'm working on this.

Question. Trying to get `debug` to work while working on this. Added the expected require and calls in the code and ran with:

`DEBUG=eslint:no-restricted-imports ./node_modules/.bin/mocha tests/lib/rules/no-restricted-imports.js`

However can't produce any debug output. Any ideas on how to enable debug output when running a test?

Here is where I'm referencing it:
https://github.com/guyellis/eslint/blob/no-restricted-imports/lib/rules/no-restricted-imports.js#L7

And here is where I'm using it:
https://github.com/guyellis/eslint/blob/no-restricted-imports/lib/rules/no-restricted-imports.js#L37

**Comment by nzakas** [comment]
Seems like that should work. Maybe try `DEBUG=eslint:*` instead? (Avoid typos)

**Comment by guyellis** [comment]
PR submitted: https://github.com/eslint/eslint/pull/4590

This is my first PR for ESLint. Let me know if it needs finessing.
